### PR TITLE
Interpreter fixes.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/interp/transform.c [hunk 1]
```diff
@@ -1648,7 +1648,9 @@ interp_handle_intrinsics (TransformData *td, MonoMethod *target_method, MonoClas
 			td->ip += 5;
 			return TRUE;
 		}
-	} else if (in_corlib && !strcmp (klass_name_space, "Internal.Runtime.CompilerServices") && !strcmp (klass_name, "Unsafe")) {
+	} else if (((in_corlib && !strcmp (klass_name_space, "Internal.Runtime.CompilerServices"))
+				|| !strcmp (klass_name_space, "System.Runtime.CompilerServices"))
```
Comment (jaykrell) [inline mono/mini/interp/transform.c]: Parens to group the `||`?
Actually it doesn't look correct anyway.
It looks like `in_corlib` is not guarding as much as it should.

File: mono/mini/interp/transform.c [hunk 2]
```diff
@@ -1708,6 +1710,26 @@ interp_handle_intrinsics (TransformData *td, MonoMethod *target_method, MonoClas
...
+			gboolean has_refs;
+
+			MonoClass *klass = mono_class_from_mono_type_internal (t);
+			mono_class_init_internal (klass);
+			if (MONO_TYPE_IS_REFERENCE (t))
+				has_refs = TRUE;
+			else if (MONO_TYPE_IS_PRIMITIVE (t))
+				has_refs = FALSE;
+			else
+				has_refs = m_class_has_references (klass) ? TRUE : FALSE;
```
Comment (jaykrell) [inline mono/mini/interp/transform.c]: Remove the ternary and just `has_refs = m_class_has_references`?

File: mono/mini/interp/transform.c (hunk 1 above)
Comment (BrzVlad) [reply mono/mini/interp/transform.c]: @jaykrell I assume on netcore these intrinsics are found also in a different assembly for public use

Comment (jaykrell) [reply mono/mini/interp/transform.c]: Yeah I wondered that, could be.
A shame to strcmp so often imho.

**Comment by lewurm** [comment]
@monojenkins build failed

**Comment by lewurm** [comment]
@monojenkins squash

**Comment by filipnavara** [comment]
@monojenkins build Linux ARMv7

**Pull request merged**